src/content/docs/en/guides/middleware.mdx

```diff
@@ -200,3 +192,7 @@ greeting response
 auth response
 validation response
 ```
+
+### Handling 404 pages
+
+Middleware will run for an on-demand rendered route while Astro attempts to render a 404 page. However, it is up to the [adapter](/en/guides/server-side-rendering/) to decide whether that code runs. Some adapters may serve a platform-specific error page instead.
```

Comment:
```suggestion
If your custom 404 page is on-demand rendered, or if astro's default 404 page is being used, middleware will run for requests that did not match a route. However, it is up to the [adapter](/en/guides/server-side-rendering/) to decide whether that code runs. Some adapters may serve a platform-specific error page instead.
```

Reply:
I feel a little hesitant using the term "default 404 page" because it is just a blank white page. There may not be any avoiding it, and that's fine too.

Reply:
I feel like my sentence is a cleaner user-facing statement -- middleware will run when...

What about something closer to:

> Middleware will attempt to run for all on-demand rendered pages, even when a matching route cannot be found. This includes Astro's default (blank) 404 page and any custom 404 pages. However, ...

Reply:
Updated your suggestion with min to commit. See what you think of this